hudi-spark-datasource/hudi-spark-common/src/main/scala/org/apache/spark/sql/hudi/streaming/HoodieStreamSource.scala

```diff
@@ -151,9 +160,9 @@ class HoodieStreamSource(
         // Consume the data between (startCommitTime, endCommitTime]
         val incParams = parameters ++ Map(
           DataSourceReadOptions.QUERY_TYPE.key -> DataSourceReadOptions.QUERY_TYPE_INCREMENTAL_OPT_VAL,
-          DataSourceReadOptions.READ_BY_STATE_TRANSITION_TIME.key() -> useStateTransitionTime.toString,
           DataSourceReadOptions.BEGIN_INSTANTTIME.key -> startCommitTime(startOffset),
-          DataSourceReadOptions.END_INSTANTTIME.key -> endOffset.commitTime
+          DataSourceReadOptions.END_INSTANTTIME.key -> endOffset.commitTime,
```

Comment:
Do we have plan to bookkeep the completion time so that we can find out the missing instants automatically?
Maybe just create a JIRA first.

Reply:
filed https://issues.apache.org/jira/browse/HUDI-6377